src/iff/data.f90

```diff
@@ -0,0 +1,155 @@
+! This file is part of xtb.
+!
+! Copyright (C) 2022 Christoph Plett
+!
+! xtb is free software: you can redistribute it and/or modify it under
+! the terms of the GNU Lesser General Public License as published by
+! the Free Software Foundation, either version 3 of the License, or
+! (at your option) any later version.
+!
+! xtb is distributed in the hope that it will be useful,
+! but WITHOUT ANY WARRANTY; without even the implied warranty of
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU Lesser General Public License for more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+!> Topological data for force field type calculations
+module xtb_iff_data
+   use xtb_mctc_accuracy, only: wp
+   implicit none
+   private
+
+   public :: TIFFData
+
+   type :: TIFFData
+
+      integer :: n !Total number atoms
+      integer :: n1 !Number atoms molA
+      integer :: n2 !Number atoms molB
+      integer :: nlmo1 !Number lmos
+      integer :: nlmo2
+      real(wp), allocatable :: xyz1(:, :) !Coordinates
+      real(wp), allocatable :: rlmo1(:, :) !LMO position
+      real(wp), allocatable :: q1(:) !Charge
+      real(wp), allocatable :: qdr1(:)
+      real(wp), allocatable ::xyzdr1(:, :)
+      real(wp), allocatable :: cn1(:)
+      real(wp), allocatable :: z1(:)
+      real(wp), allocatable :: alp1(:)
+      real(wp), allocatable :: qct1(:, :)
+      integer, allocatable :: at1(:)
+      integer, allocatable :: lmo1(:)
+      real(wp), allocatable :: xyz2(:, :)
+      real(wp), allocatable :: rlmo2(:, :)
+      real(wp), allocatable :: q2(:)
+      real(wp), allocatable :: qdr2(:)
+      real(wp), allocatable ::xyzdr2(:, :)
+      real(wp), allocatable :: cn2(:)
+      real(wp), allocatable :: z2(:)
+      real(wp), allocatable :: alp2(:)
+      real(wp), allocatable :: qct2(:, :)
+      integer, allocatable :: at2(:)
+      integer, allocatable :: lmo2(:)
+      real(wp), allocatable :: c6ab(:, :)
+      real(wp), allocatable :: alpab(:, :)
+      real(wp), allocatable :: cprob(:)
+      real(wp), allocatable :: xyz(:, :)
+      real(wp), allocatable :: q(:)
+      real(wp), allocatable :: cn(:)
+      real(wp), allocatable :: alp0(:)
+      real(wp), allocatable :: gab1(:, :)
+      real(wp), allocatable :: gab2(:, :)
+      real(wp), allocatable :: den1(:, :, :)
+      real(wp), allocatable :: den2(:, :, :)
+      real(wp), allocatable :: qcm1(:)
+      real(wp), allocatable :: qcm2(:)
+      integer, allocatable :: at(:)
+      integer, allocatable :: neigh(:, :)
+
+   contains
+
+      procedure :: zero
+      procedure :: allocateIFFData
+
+   end type TIFFData
+
+contains
+
+   subroutine zero(self)
```

Comment:
This name is misleading, it deallocates the structure rather than setting it to zero. Consider `delete`, `destroy`, or `free`.